client/request.go

```diff
@@ -278,7 +278,14 @@ DoneChoosingBodySource:
 	if r.pathPattern != "" && r.pathPattern != "/" && r.pathPattern[len(r.pathPattern)-1] == '/' {
 		reinstateSlash = true
 	}
-	urlPath := path.Join(basePath, r.pathPattern)
+
+	basePathURL, err := url.Parse(basePath)
+	if err != nil {
+		return nil, err
+	}
+	basePathQueryParams := basePathURL.Query()
```

Comment:
Can you add a comment explaining the logic here? Which one takes priority etc.

Reply:
Done